bitsandbytes/backends/_subinterfaces.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+
+
+class RequiredUtilities(ABC):
+    @abstractmethod
+    def check_matmul():
+        raise NotImplementedError
+
+
+class FourBitMatmul(ABC):
```

Comment:
Something to think about here is that for the nested quantization, the interface for `KBitQuantization` is needed (quantize_blockwise/dequantize_blockwise)